luxonis_ml/nn_archive/config_building_blocks/base_models/input.py

```diff
@@ -66,39 +68,33 @@ class Input(BaseModelExtraForbid):
     input_type: InputType = Field(description="Type of input data (e.g., 'image').")
     shape: List[int] = Field(
         min_length=1,
-        max_length=5,
         description="Shape of the input data as a list of integers (e.g. [H,W], [H,W,C], [N,H,W,C], ...).",
     )
-    layout: str = Field(
-        description="Lettercode interpretation of the input layout (e.g., 'NCHW').",
-        default="NCHW",
+    layout: List[str] = Field(
```

Comment:
If `parse_layout` accepts a `str` as well, should this type be `Union[List[str], str]`?

Reply:
The output of `parse_layout` is always `List[str]` which is what will be assigned to the `layout` field.